### ISSUE TITLE
Small tweaks to pool implementation

### ISSUE THREAD

**Issue by ignopeverell** [ignopeverell/grin#59]
A couple things that I missed previously on the pool implementation from @MoaningMyrtle that should be updated:

* A few methods (like adding transaction to the pool) are "&mut self". As there's only one pool per node and it will have to be a shared reference, it needs to have full interior mutability. I believe only transactions HashMap is posing an issue and can be protected by a mutex.
* What right now is the DummyChain trait has get_best_utxo_set method, which basically returns a map of UTXOs. On that structure, the pool only ever uses get_output(commitment). It'd be easier for implementers of the trait to just have a get_unspent(commitment), without the set implementation. This way they wouldn't have to provide some set wrapper or view.

@MoaningMyrtle let me know if that makes sense. Thanks!

**Comment by MoaningMyrtle** [comment]
Makes sense to me! Figured that these things will come up as integration proceeds. I'll take a crack at these things ASAP.

**Comment by MoaningMyrtle** [comment]
Re: point 1, I think it'd be best to wrap the whole pool in a RwLock in those cases (so an `Arc<RwLock<TransactionPool>>` or similar construction), as the pool itself operates under the assumption of exclusive access to its two mut entrypoints in order to ensure internal consistency. Does this make sense?

Re: point 2, this totally makes sense, I'll prepare a PR.

**Comment by ignopeverell** [comment]
Sounds good for point 1. Thanks!
